src/e2e/java/teammates/e2e/util/LNPResultsStatistics.java

```diff
@@ -0,0 +1,68 @@
+package teammates.e2e.util;
+
+import org.testng.Reporter;
+
+/**
+ * Represents the L&P test results statistics.
+ */
+public class LNPResultsStatistics {
+
+    private double errorPct;
+    private double meanResTime;
+    private double pct1ResTime; // 90th percentile
+    private double throughput;
+    private int errorCount;
+    private int sampleCount;
+    private String resultsErrorMessage;
+
+    /**
+     * Verify the LNP results statistics with the specified threshold.
+     */
+    public void verifyLnpTestSuccess(double errorRateLimit, double meanResTimeLimit) {
+        checkErrorLimit(errorRateLimit);
+        checkMeanResTimeLimit(meanResTimeLimit);
+        if (resultsErrorMessage != null) {
```

Comment:
Your initial approach was the better choice here 😅 . Just noticed that we're appending to a null string in the methods below. Let's initialise the string to `""` and check `!resultsErrorMessage.isEmpty()` here?